tests/apollo/test_skvbc.py

```diff
@@ -267,6 +267,30 @@ async def test_delayed_replicas_start_up(self, bft_network, skvbc, nursery):
                     self.assertTrue(last_block > current_block)
                     current_block = last_block
 
+            current_primary = current_view % bft_network.config.n
+            non_primaries = bft_network.all_replicas(without={current_primary})
+            restarted_replica = random.choice(non_primaries)
+            print(f"Restart replica #{restarted_replica} to make sure its view is persistent...")
+
+            bft_network.stop_replica(restarted_replica)
+            await trio.sleep(seconds=5)
+            bft_network.start_replica(restarted_replica)
+
+            await trio.sleep(seconds=20)
+
+            # The view is stable, stop sending requests
+            nursery.cancel_scope.cancel()
+
+            key = ['replica', 'Gauges', 'lastExecutedSeqNum']
+            last_executed_seq_num = await bft_network.metrics.get(current_primary, *key)
+
+            with trio.fail_after(seconds=30):
+                while True:
+                    last_executed_seq_num_restarted = await bft_network.metrics.get(restarted_replica, *key)
+                    if last_executed_seq_num_restarted >= last_executed_seq_num:
```

Comment:
Why would this almost not always be equivalent, considering you stop the nursery before sending the get request for the metric. I see there is a race, but it seems like more likely than not progress might have already stopped by the time we get here.

In other words, what's the purpose of this check?

Reply:
Hi Andrew, the idea is to check that the restarted replica is in the new view after the restart, and executing requests.